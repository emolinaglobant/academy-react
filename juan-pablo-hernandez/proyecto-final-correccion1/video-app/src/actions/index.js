export const setFavorite = payload => ({
  type: 'SET_FAVORITE',
  payload,
});

export const deleteFavorite = payload => ({
  type: 'DELETE_FAVORITE',
  payload,
});

export const loginRequest = payload => ({
  type: 'LOGIN_REQUEST',
  payload,
});

export const logoutRequest = payload => ({
  type: 'LOGOUT_REQUEST',
  payload,
});

export const registerRequest = payload => ({
  type: 'REGISTER_REQUEST',
  payload,
});

export const getVideoSource = payload => ({
  type: 'GET_VIDEO_SOURCE',
  payload,
});
