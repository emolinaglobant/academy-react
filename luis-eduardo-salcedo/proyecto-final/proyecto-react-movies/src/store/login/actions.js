import * as types from "./actionTypes";
import { auth, googleAuthProvider, facebookAuthProvider } from "../../firebase";
const registerStart = () => ({
    type: types.REGISTER_START,
});
const registerSuccess = (user) => ({
    type: types.REGISTER_SUCCESS,
    payload: user,
});
export const registerFail = (error) => ({
    type: types.REGISTER_FAIL,
    payload: error,
});

const loginStart = () => ({
    type: types.LOGIN_START,
});
const loginSuccess = (user) => ({
    type: types.LOGIN_SUCCESS,
    payload: user,
});
export const loginFail = (error) => ({
    type: types.LOGIN_FAIL,
    payload: error,
});

const logoutStart = () => ({
    type: types.LOGOUT_START,
});
const logoutSuccess = () => ({
    type: types.LOGOUT_SUCCESS,
});
const logoutFail = (error) => ({
    type: types.LOGOUT_FAIL,
    payload: error,
});

export const setUser = (user) => ({
    type: types.SET_USER,
    payload: user,
});

const googleSignInStart = () => ({
    type: types.GOOGLE_SIGN_IN_START,
});
const googleSignInSuccess = (user) => ({
    type: types.GOOGLE_SIGN_IN_SUCCESS,
    payload: user,
});
const googleSignInFail = (error) => ({
    type: types.GOOGLE_SIGN_IN_FAIL,
    payload: error,
});

const facebookSignInStart = () => ({
    type: types.FACEBOOK_SIGN_IN_START,
});
const facebookSignInSuccess = (user) => ({
    type: types.FACEBOOK_SIGN_IN_SUCCESS,
    payload: user,
});
const facebookSignInFail = (error) => ({
    type: types.FACEBOOK_SIGN_IN_FAIL,
    payload: error,
});

function registerInitiate(email, password, displayName) {
    return function (dispatch) {
        dispatch(registerStart());
        auth.createUserWithEmailAndPassword(email, password)
            .then(({ user }) => {
                user.updateProfile({
                    displayName,
                });
                dispatch(registerSuccess(user));
            })
            .catch((error) => dispatch(registerFail(error.message)));
    };
}

export const loginInitiate = (email, password) => {
    return function (dispatch) {
        dispatch(loginStart());
        auth.signInWithEmailAndPassword(email, password)
            .then(({ user }) => {
                dispatch(loginSuccess(user));
            })
            .catch((error) => dispatch(loginFail(error.message)));
    };
};

export const logoutInitiate = () => {
    return function (dispatch) {
        dispatch(logoutStart());
        auth.signOut()
            .then((resp) => {
                dispatch(logoutSuccess());
            })
            .catch((error) => dispatch(logoutFail(error.message)));
    };
};

export const googleSignInInitiate = () => {
    return function (dispatch) {
        dispatch(googleSignInStart());
        auth.signInWithPopup(googleAuthProvider)
            .then(({ user }) => {
                dispatch(googleSignInSuccess(user));
            })
            .catch((error) => dispatch(googleSignInFail(error.message)));
    };
};

export const facebookSignInInitiate = () => {
    return function (dispatch) {
        dispatch(facebookSignInStart());
        auth.signInWithPopup(
            facebookAuthProvider.addScope("user_birthday, email")
        )
            .then(({ user }) => {
                dispatch(facebookSignInSuccess(user));
            })
            .catch((error) => dispatch(facebookSignInFail(error.message)));
    };
};

export default registerInitiate;
