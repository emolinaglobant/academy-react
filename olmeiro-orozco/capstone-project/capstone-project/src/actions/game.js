import { getQuestions } from "../helpers/getQuestions";
import { types } from "../types/types";

export const startRound = (start, numberCategory, difficulty) => {
  return {
    type: types.startRound,
    payload: {
      start: start,
      numberCategory: numberCategory,
      difficulty: difficulty,
    },
  };
};

export const loadQuestions = () => {
  return async (dispatch, getState) => {
    const { numberCategory, difficulty } = getState().game;
    const data = await getQuestions(numberCategory, difficulty);

    dispatch(storeQuestions(data));
  };
};

export const storeQuestions = (questions) => {
  return {
    type: types.gameStoreQuestions,
    payload: {
      questions,
    },
  };
};

export const increasePrize = (increase) => {
  return {
    type: types.gameIncreasePrize,
    payload: {
      totalPrize: increase,
    },
  };
};

export const addRound = (increase) => {
  return {
    type: types.gameAddRound,
    payload: {
      round: increase,
    },
  };
};

export const roundChange = (value) => {
  return {
    type: types.gameChangeRound,
    payload: {
      changeRound: value,
    },
  };
};

export const resetGame = () => {
  return {
    type: types.gameReset,
  };
};
