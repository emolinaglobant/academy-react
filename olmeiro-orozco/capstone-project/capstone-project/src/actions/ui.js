import { types } from "../types/types";

export const setError = (err) => {
  return {
    type: types.uiSetError,
    payload: err,
  };
};

export const removeError = (err) => {
  return {
    type: types.uiRemoveError,
  };
};

export const startLoading = () => ({
  type: types.uiStartLoading,
});

export const finishLoading = () => ({
  type: types.uiFinishLoading,
});
