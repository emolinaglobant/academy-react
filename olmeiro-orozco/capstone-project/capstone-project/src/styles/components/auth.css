.auth__main {
  align-items: center;
  background-color: var(--BlueHaze);
  background-image: linear-gradient(rgba(0, 0, 255, 0.5), rgba(255, 255, 0, 0.5)), url("../../../public/assets/trivia2.jpg");
  display: flex;
  height: 100vh;
  justify-content: center;
  margin: 0px;
  width: 100vw;
}

.auth__box-container {
  background-color: var(--white);
  border-radius: 2px;
  box-shadow: 0px 3px var(--CopperRose);
  padding: 20px;
  width: 250px;
}

.auth__title {
  color: var(--darkgray);
  margin-bottom: 10px;
}

.auth__input {
  border-bottom: 1px solid var(--darkgray);
  border: 0px;
  color: var(--darkgray);
  font-size: 16px;
  height: 20px;
  margin-bottom: 20px;
  transition: border-bottom .3s ease-out;
  width: 100%;
}

.auth__input:focus {
  outline: none;
  border-bottom: 1px solid var(--primary);
}

.auth__google {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-bottom: 20px;
  padding-top: 20px;
  width: 100%;
}

.auth__alert-error {
  background-color: var(--red);
  border-radius: 5px;
  color: var(--secondary);
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}