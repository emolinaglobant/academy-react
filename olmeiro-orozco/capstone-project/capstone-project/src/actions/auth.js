import { createUserWithEmailAndPassword, getAuth, signInWithEmailAndPassword, signInWithPopup, signOut, updateProfile } from "firebase/auth";
import { googleAuthProvider } from "../firebase/firebase-config";
import { types } from "../types/types";
import { finishLoading, startLoading } from "./ui";
import Swal from "sweetalert2";

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());

    const auth = getAuth();
    signInWithEmailAndPassword(auth, email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));

        dispatch(finishLoading());
      })
      .catch((e) => {
        dispatch(finishLoading());
        Swal.fire({
          icon: 'error',
          title: 'Oops...',
          text: 'Something went wrong!',
          footer: 'Please confirm your Email and Password.'
        })
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    const auth = getAuth();
    createUserWithEmailAndPassword(auth, email, password)
      .then(async ({ user }) => {
        await updateProfile(user, { displayName: name });

        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        console.log(error);
        Swal.fire("Error", error.message, "error");
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    const auth = getAuth();
    signInWithPopup(auth, googleAuthProvider).then(({ user }) => {
      dispatch(login(user.uid, user.displayName));
    });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const startLogout = () => {
  return async (dispath) => {
    const auth = getAuth();
    await signOut(auth);

    dispath(logout());
  };
};

export const logout = () => {
  return {
    type: types.logout,
  };
};
