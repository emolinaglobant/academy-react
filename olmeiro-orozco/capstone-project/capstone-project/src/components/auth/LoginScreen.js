import React from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { startGoogleLogin, startLoginEmailPassword } from "../../actions/auth";

import { useForm } from "../../hooks/useForm";

export const LoginScreen = () => {
  const dispatch = useDispatch();
  const { loading } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    email: "",
    password: "",
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(startLoginEmailPassword(email, password));
  };

  const handleGoogleLogin = () => {
    dispatch(startGoogleLogin());
  };

  return (
    <section>
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin}>
        <input
          autoComplete="off"
          className="auth__input"
          name="email"
          onChange={handleInputChange}
          placeholder="email"
          type="text"
          value={email}
        />
        <input
          className="auth__input"
          name="password"
          onChange={handleInputChange}
          placeholder="password"
          type={"password"}
          value={password}
        />

        <button className="btn btn__primary" disabled={loading} type="submit">
          Login
        </button>

        <section className="auth__google">
          <p>Login with google</p>

          <section className="google__btn" onClick={handleGoogleLogin}>
            <section className="google__icon-wrapper">
              <img
                alt="google button"
                className="google__icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
              />
            </section>
            <p className="btn__text">
              <b>Sign in with google</b>
            </p>
          </section>
        </section>
        <Link to="/auth/register" className="link">
          Create new account
        </Link>
      </form>
    </section>
  );
};
