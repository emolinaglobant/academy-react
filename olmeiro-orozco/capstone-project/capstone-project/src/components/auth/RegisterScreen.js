import React from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import validator from "validator";
import { removeError, setError } from "../../actions/ui";
import { startRegisterWithEmailPasswordName } from "../../actions/auth";
import { useSelector, useDispatch } from "react-redux";

export const RegisterScreen = () => {
  const dispatch = useDispatch();
  const { msgError } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    name: "",
    email: "",
    password: "",
    password2: "",
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name));
    }
  };

  const isFormValid = () => {
    if (validator.isEmpty(name)) {
      dispatch(setError("name is required"));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError("Email is required"));
      return false;
    } else if (password !== password2 || password.length < 5) {
      dispatch(
        setError(
          "passwor should be at least six characters and match each other"
        )
      );
      return false;
    }
    dispatch(removeError());
    return true;
  };

  return (
    <section>
      {msgError && <section className="auth__alert-error">{msgError}</section>}

      <h3 className="auth__title">Register</h3>

      <form onSubmit={handleRegister}>
        <input
          className="auth__input"
          type="text"
          placeholder="name"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          value={email}
          onChange={handleInputChange}
          type="email"
          placeholder="email"
          name="email"
          autoComplete="off"
        />
        <input
          className="auth__input"
          type={"password"}
          placeholder="password"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type={"password"}
          placeholder="confirm password"
          name="password2"
          value={password2}
          onChange={handleInputChange}
        />

        <button type="submit" className="btn btn__primary">
          Register
        </button>

        <Link to="/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </section>
  );
};
