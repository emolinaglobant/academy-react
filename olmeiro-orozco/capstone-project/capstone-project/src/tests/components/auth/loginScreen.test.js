import React from "react";
import { LoginScreen } from "../../../components/auth/LoginScreen";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { startGoogleLogin, startLoginEmailPassword } from "../../../actions/auth";

jest.mock("../../../actions/auth", () => ({
  startGoogleLogin: jest.fn(),
  startLoginEmailPassword: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null,
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <LoginScreen />
    </MemoryRouter>
  </Provider>
);

describe("Tests on <LoginScreen /> ", () => {
  beforeEach(() => {
    store = mockStore(initState);
    jest.clearAllMocks();
  });

  test("should show it correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should dispatch action startGoogleLogin", () => {
    wrapper.find(".google__btn").prop("onClick")();
    expect(startGoogleLogin).toHaveBeenCalled();
  });

  test("should dispatch action startLoginEmailPassword", () => {
    wrapper.find("form").prop("onSubmit")({ preventDefault() {} });
    expect(startLoginEmailPassword).toHaveBeenCalled();
  });

  test("should dispatch action startLoginEmailPassword", () => {
    wrapper.find("form").prop("onSubmit")({ preventDefault() {} });
    expect(startLoginEmailPassword).toHaveBeenLastCalledWith("","");
  });
});
