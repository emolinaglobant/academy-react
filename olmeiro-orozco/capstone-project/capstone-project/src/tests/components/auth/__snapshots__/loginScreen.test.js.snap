// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <LoginScreen />  should show it correctly 1`] = `
<section>
  <h3
    className="auth__title"
  >
    Login
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="btn btn__primary"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <section
      className="auth__google"
    >
      <p>
        Login with google
      </p>
      <section
        className="google__btn"
        onClick={[Function]}
      >
        <section
          className="google__icon-wrapper"
        >
          <img
            alt="google button"
            className="google__icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </section>
        <p
          className="btn__text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </section>
    </section>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Create new account
    </a>
  </form>
</section>
`;
